akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -87,4 +92,38 @@ class FusingSpec extends AkkaSpec with ConversionCheckedTripleEquals {
 
   }
 
+  "SubFusingActorMaterializer" must {
+
+    "work with asynchronous boundaries in the subflows" in {
+      val async = Flow[Int].map(_ * 2).withAttributes(Attributes.asyncBoundary)
```

Comment:
we need a sugar method for this. Maybe Async(Graph): Graph, and it probably should fuse by default.
We can even go further and have `.viaAsync(Flow[Int].map(_ * 2))`.